### PR TITLE
Integrate `bpmn-js-tracking`

### LINKED ISSUE #3572: Integrate `bpmn-js-tracking`
### What should we do?

We should integrate `bpmn-js-tracking`.

Breakdown:
* Integrate `bpmn-js-tracking` with Mixpanel
* Filter and clean data from tracking framework: agree on important information to include for ModdleElements 

### Why should we do it?

Part of https://github.com/bpmn-io/internal-docs/issues/746

### PR DESCRIPTION
**Pull request by smbea**
Closes #3572 

This PR integrates `bpmn-js-tracking` by:
 * including modules in platform and cloud modelers
 * adding `ModelingEventHandler` (similar to existing tracking) that handles the `bpmn-js-tracking` events
 * transforming `bpmn-js-tracking` events data. Only send relevant information for elements: currently id, element type, template applied. We can expand this as needed (cc @christian-konrad )
 
While doing this I found these issues with `bpmn-js-tracking`, but we can start with this review and just bump the version when these are implemented:
 * https://github.com/bpmn-io/bpmn-js-tracking/issues/21 
 * https://github.com/bpmn-io/bpmn-js-tracking/issues/22

### CODE REVIEW

**Comment by smbea** [comment]
@nikku Would love your opinion on the architectural side of this/how it fits into desktop modeler/how this could be applied to desktop modeler. Happy to discuss this in person if easier 🙂

File: client/src/plugins/user-journey-statistics/event-handlers/ModelingEventHandler.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      });
+    });
+  };
+
+}
+
+
+// helpers //////////
+
+function replaceModleElement(object) {
```
Comment (nikku) [inline client/src/plugins/user-journey-statistics/event-handlers/ModelingEventHandler.js:68]:
```suggestion
function replaceModdleElement(object) {
```

Are these `moddle` (businessObject) or diagram elements?

**Review by nikku — COMMENTED** [review]
Looks great!

File: client/src/plugins/user-journey-statistics/event-handlers/ModelingEventHandler.js (hunk 1 above)
Comment (smbea) [reply client/src/plugins/user-journey-statistics/event-handlers/ModelingEventHandler.js:68]: Right now it's the whole diagram element; we send `element: {id, type, template?}`. We may decide to extend the data we send in the future (or now already if we see fit), for example considering event definitions and such. I decided to keep it short and expand as needed. Also why @christian-konrad is cc'ed here.

Comment (nikku) [reply client/src/plugins/user-journey-statistics/event-handlers/ModelingEventHandler.js:68]: Let's rename this method to `filterDiagramElement` then or something.

File: client/src/app/tabs/cloud-bpmn/modeler/BpmnModeler.js [hunk 2]
```diff
@@ -18,6 +18,11 @@ import handToolOnSpaceModule from '../../bpmn/modeler/features/hand-tool-on-spac
 import propertiesPanelKeyboardBindingsModule from '../../bpmn/modeler/features/properties-panel-keyboard-bindings';
 import lintingAnnotationsModule from '@camunda/linting/modeler';
 
+import {
+  BpmnJSTracking as bpmnJSTracking,
+  BpmnJSTrackingModules as bpmnJSTrackingModules
```
Comment (nikku) [inline client/src/app/tabs/cloud-bpmn/modeler/BpmnModeler.js:23]: My favorite would actually be to explicitly expose different tracking modules here, and include them.

This way it is clear "what we use". But let's get this integrated for now and consider moving to explicit tracking features in the future.

File: client/src/plugins/user-journey-statistics/event-handlers/ModelingEventHandler.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
```
Comment (nikku) [inline client/src/plugins/user-journey-statistics/event-handlers/ModelingEventHandler.js:72]: There is still a typo in here which we want to fix.

If we replace a diagram element, let's call this `replaceDiagramElement`, if we replace a `ModdleElement` (`x.businessObject`), let's call this `replaceModdleElement`.

Comment (smbea) [reply client/src/plugins/user-journey-statistics/event-handlers/ModelingEventHandler.js]: My bad, I thought I had replaced this already

File: client/src/app/tabs/cloud-bpmn/modeler/BpmnModeler.js (hunk 2 above)
Comment (nikku) [reply client/src/app/tabs/cloud-bpmn/modeler/BpmnModeler.js:23]: I think both would work. If we go the import `lib` path I'd probably rename it from `trackingModules` to `tracking` or even better `features`.

So `'bpmn-js-tracking/lib/trackingModules/popupMenu'` becomes `'bpmn-js-tracking/lib/features/popup-menu'`;

**Review by nikku — APPROVED** [review]
Good stuff.

**Pull request merged**